syntax = "proto3";

package inference;

service InferenceService {
  rpc Predict(InferenceRequest) returns (InferenceResponse);
}

message InferenceRequest {
  repeated float tensor_data = 1;
}

message InferenceResponse {
  string predicted_class = 1;
  float confidence = 2;
}
